### DISCUSSION TITLE
Graph Node highlight ports

### DISCUSSION THREAD

**Discussion by Zireael07** [godotengine/godot-proposals#5249] [GUI]
Ports in GraphNode can be difficult to see/click at some zoom levels/font sizes.

https://user-images.githubusercontent.com/83387915/122633500-55bf1980-d0d9-11eb-9772-d033f8bd779b.gif

(taken from closed proposal which mentioned Visual Scripting but this applies to all Graph Nodes

**Comment by Calinou** [comment]
I don't understand what's supposed to happen when dragging a node. Can you explain?

**Comment by Zireael07** [comment]
Literally what the gif shows - the port is highlighted (bigger and/or in a different color)

**Comment by maximkulkin** [comment]
I guess, the proper description should be as follows:

Sometimes GraphNode ports are hard to click on and there is no visual indication if clicking and dragging from a port will start making connection rather than dragging the whole node.

Proposal:

Make it easier to know if port is hovered over. Add theme settings for highlighted (hovered over) GraphNode ports that will take effect when mouse cursor hovers over them. Default styles should implement slight increase in port size as visual indication of port being highlighted.
